src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -51,6 +54,21 @@ public static Name parseName(String name) throws ParseException {
         }
         return new Name(trimmedName);
     }
+    /**
+     * Parses a {@code String name} into a {@code Name}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code name} is invalid.
+     */
+    public static ContactName parseContactName(String name) throws ParseException {
+        requireNonNull(name);
+        String trimmedName = name.trim();
+        if (!ContactName.isValidName(trimmedName)) {
+            throw new ParseException(ContactName.MESSAGE_CONSTRAINTS);
```

Comment:
Just merged my ParserUtil.java which also implements these functions. Can solve merge conflicts to ensure parser for ContactName, Phone, Email throws ParseException in the form:  throw new ParseException(String.format("<field> given: %s\n%s", trimmed<field>, <field>.MESSAGE_CONSTRAINTS));